### PR TITLE
CI-9639: added github_app support in http  credentials

### PR DESCRIPTION
**Pull request by sahithibanda01**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
add test in github_test.go and test it pointing to qa and prod

File: examples/resources/harness_platform_connector_github/resource.tf [hunk 1]
```diff
@@ -61,3 +61,31 @@ resource "harness_platform_connector_github" "test" {
     }
   }
 }
+
+resource "harness_platform_connector_github" "test" {
```
Comment (rathodmeetsatish) [inline examples/resources/harness_platform_connector_github/resource.tf:65]: changes in example folder will require doc update

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/antihax/optional v1.0.0
 	github.com/docker/docker v24.0.5+incompatible
-	github.com/harness/harness-go-sdk v0.3.49
+	github.com/harness/harness-go-sdk v0.3.52-0.20230926100809-9d31d2a3e28f
```
Comment (smjt-h) [inline go.mod:8]: Lets test after importing latest sdk

**Comment by smjt-h** [comment]
Validate support for datasource as well `harness_platform_connector_github`

**Review by smjt-h — APPROVED** [review]

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
Add changelog file corresponding to your pr number

File: .changelog/703.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+```release-note:added new feature in a resource
```
Comment (rathodmeetsatish) [inline .changelog/703.txt:1]: this is enhancement please refer to some file where we have done enchancement and use the same template

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

File: examples/resources/harness_platform_connector_github/resource.tf (hunk 1 above)
Comment (autociuser) [reply examples/resources/harness_platform_connector_github/resource.tf:65]: done

Comment (sahithibanda01) [reply examples/resources/harness_platform_connector_github/resource.tf:65]: done

**Pull request merged**